xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp

```diff
@@ -121,7 +121,38 @@ bool CLogindUPowerSyscall::HasLogind()
   // recommended method by systemd devs. The seats directory
   // doesn't exist unless logind created it and therefore is running.
   // see also https://mail.gnome.org/archives/desktop-devel-list/2013-March/msg00092.html
-  return (access("/run/systemd/seats/", F_OK) >= 0);
+  if (access("/run/systemd/seats/", F_OK) >= 0)
+    return true;
+
+  // on some environments "/run/systemd/seats/" doesn't exist, e.g. on flatpak. Try DBUS instead.
+  CDBusMessage message(LOGIND_DEST, LOGIND_PATH, LOGIND_IFACE, "ListSeats");
+  DBusMessage *reply = message.SendSystem();
+  if (!reply)
+    return false;
+
+  DBusMessageIter arrIter;
+  dbus_message_iter_init(reply, &arrIter);
+  if (dbus_message_iter_get_arg_type(&arrIter) == DBUS_TYPE_ARRAY)
+  {
+    DBusMessageIter structIter;
+    dbus_message_iter_recurse(&arrIter, &structIter);
+    if (dbus_message_iter_get_arg_type(&structIter) == DBUS_TYPE_STRUCT)
+    {
+      DBusMessageIter strIter;
+      dbus_message_iter_recurse(&structIter, &strIter);
+      if (dbus_message_iter_get_arg_type(&strIter) == DBUS_TYPE_STRING)
+      {
+        char *seat;
+        dbus_message_iter_get_basic(&strIter, &seat);
+        if (StringUtils::StartsWith(seat, "seat"))
```

Comment:
Why is this necessary?

Reply:
it isn't absolutely necessary, we could probably get away with just testing if the dbus call returns != nullptr, but I wanted to be precise and check if we really get a seat back.
We could relax this once we have some wider testing.

Reply:
IMO since we're already calling `ListSeats`, it would be sufficient to call `dbus_message_iter_get_array_length` and be happy if that is non-zero

Reply:
Not a blocker as long as you promise to remove it later ;-)
Depending on the results of course.